### PR TITLE
Parsers: add reStructuredText

### PR DESCRIPTION
**Pull request by stsewd**
Hi everyone, thanks for such great work with this plugin, I'm really enjoying it!

I'm writing a rst grammar (<https://github.com/stsewd/tree-sitter-rst>), still wip, so feel free to not merge this till it's more complete (if you merge this now I'll make sure to keep the queries updated).

A screenshot with the current state

### Without treesitter

Screenshot from 2020-07-26 11-02-28

### With treesitter
Screenshot from 2020-07-26 11-03-33

I'm not sure about some highlight groups, took some inspiration from https://github.com/neovim/neovim/blob/master/runtime/syntax/rst.vim

### CODE REVIEW

File: doc/nvim-treesitter.txt [hunk 1]
```diff
@@ -165,7 +165,7 @@ get_node_text(node, bufnr)                        *ts_utils.get_node_text*
     return the text content of a node
 
 is_parent(dest, source)                           *ts_utils.is_parent*
-    determines wether `dest` is a parent of `source` 
+    determines wether `dest` is a parent of `source`
```
Comment (theHamsta) [inline doc/nvim-treesitter.txt:168]: whether ?

Comment (stsewd) [reply doc/nvim-treesitter.txt:168]: That was already there, only the trailing space was removed, I can fix that typo here if that's ok

Comment (theHamsta) [reply doc/nvim-treesitter.txt:168]: As you prefer, yes we placed really a lot of typos in the docs :stuck_out_tongue:

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -122,6 +122,21 @@ are optional and will not have any effect for now.
...
+
+```
+@text
+@text.strong
+@text.emphasis
+@text.underline
+@text.title
+@text.literal
+@text.uri
+```
```
Comment (vigoux) [inline CONTRIBUTING.md:138]: Awesome idea

**Pull request merged**